python/BioSimSpace/Sandpit/Exscientia/FreeEnergy/_restraint_search.py

```diff
@@ -997,71 +1189,137 @@ def getDistance(idx1, idx2, u):
             )[2][0]
             return distance
 
-        def getAngle(idx1, idx2, idx3, u):
-            """Angle in radian."""
+        def _getAngle(idx1, idx2, idx3, u):
+            """Angle in radians."""
             C = u.atoms[idx1].position
             B = u.atoms[idx2].position
             A = u.atoms[idx3].position
-            BA = A - B
-            BC = C - B
-            angle = _np.arccos(_np.dot(BA, BC) / (_norm(BA) * _norm(BC)))
+            angle = _calc_angles(C, B, A, box=u.dimensions)
             return angle
 
-        def getDihedral(idx1, idx2, idx3, idx4, u):
-            """Dihedral in radian."""
+        def _getDihedral(idx1, idx2, idx3, idx4, u):
+            """Dihedral in rad"""
             positions = [u.atoms[idx].position for idx in [idx1, idx2, idx3, idx4]]
             dihedral = _calc_dihedrals(
-                positions[0],
-                positions[1],
-                positions[2],
-                positions[3],
-                box=u.dimensions,
+                positions[0], positions[1], positions[2], positions[3], box=u.dimensions
```

Comment:
I think you could just do `sum(u.atoms[idx] for idx in [idx1, idx2, idx3, idx4]).dihedral.value()`

Reply:
Thanks

Reply:
I tried replacing with the above code and this caused many of the restraint search tests to fail due to the selection of different restraints.

Reply:
Thanks, yes that was the issue (I was assuming that `calc_angles` and `calc_dihedrals` would return the same units as your code given they're all MDAnalysis functions).